test/e2e-go/features/incentives/mining_test.go

```diff
@@ -86,16 +91,33 @@ func TestBasicPayouts(t *testing.T) {
 		block, err := client.BookkeepingBlock(status.LastRound)
 		a.NoError(err)
 
-		fmt.Printf(" 1 block %d proposed by %v\n", status.LastRound, block.Proposer)
+		fmt.Printf("block %d proposed by %v\n", status.LastRound, block.Proposer())
 		a.Zero(block.ProposerPayout()) // nobody is eligible yet (hasn't worked back to balance round)
-		a.EqualValues(5_000_000, block.Bonus.Raw)
-		fixture.WaitForRoundWithTimeout(status.LastRound + 1)
+		a.EqualValues(bonus1, block.Bonus.Raw)
+
+		// all nodes agree the proposer proposed. The paranoia here is
```

Comment:
this is a great explanation/comment

Reply:
It's not quite right anymore though. The fixes to make the interface more explicit change some of the names of things, I'll update the comment after merging the final version of that work. Leaving unresolved until then.

Reply:
I reworded to explain.  This problem shouldn't be possible any more.